directord/server.py

```diff
@@ -853,6 +853,10 @@ def worker_run(self):
                 self.thread(name="run_backend", target=self.run_backend),
                 True,
             ),
+            (
+                self.thread(name="driver", target=self.driver.run),
+                True,
+            ),
```

Comment:
need to add the same to client.py as well

Reply:
it doesn't seem like i need to run this on the client side, testing on three nodes I'm able to receive heartbeats across the network without it.